### ISSUE TITLE
How to select best training images from train set?

### ISSUE THREAD

**Issue by khawar-islam** [ultralytics/ultralytics#21542]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have a dataset that contains 170K images and all images are extracted from videos and each frame represent similar classes just little change in angle of the camera. I believe its not worthy to use all images for training and same for test set.

I used active learning approach for select best images but it did not work maybe lack of understanding.

FYI, I have images with labels how i can make automated way to select the best training images.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @khawar-islam, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response. An Ultralytics engineer will also assist you here soon 😊.

**Comment by Y-T-G** [comment]
You should not extract frames in small intervals. Otherwise you will have redundant frames like you do now.

For removing redundant images, active learning is typically the technique used. But you should first resolve the issue at the source and not extract frames at short intervals.

**Comment by khawar-islam** [comment]
Hi @Y-T-G 

Thanks for your reply. Its already done not by me but some previous worker and now i have to resolve this issue. I cannot visit the 170k images and delete them. I have the images and their labels

What active learning? please provide me any useful link i implemented simple active learning method DiNOV2+Cosine Similarity it didn't work me. I want to do it automatic to make a new subset from  from 170k samples. I am very frustrated as I have checked lots of solution.

**Comment by hannaliavoshka** [comment]
As a fast solution, you can select a desired quantity of images randomly and train the model with this subset. For instance, 17K instead of 170K, or even less (or more, depending on your needs and computational resources). This subset will not necessarily be a compilation of the 'best' images, but it will reflect the diversity of the dataset, decreasing the repetitive frames. For this stage, you can also check the number of background images (without the labelled objects) in the subset and control its quantity. Use these general tips to create a subset of the training data. I know this is not the answer to your question, but sometimes any solution as a starting point is better than no one) https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/#dataset

**Comment by glenn-jocher** [comment]
For your specific case with 170K redundant video frames, I'd recommend checking out our [active learning guide](https://www.ultralytics.com/blog/active-learning-speeds-up-computer-vision-development) which explains techniques for selecting the most informative samples from large datasets. Alternatively, you can use YOLO11's built-in subset training by setting `fraction=0.1` during training to automatically use only 10% of your data (or adjust the fraction as needed) - this provides a quick automated solution while you work on implementing more sophisticated selection methods.
